### PR TITLE
Conn edituser

### PR DESCRIPTION
**Pull request by grbarker**
Connect the EditUser component to the backend.

Addresses isssue #747

### CODE REVIEW

File: src/frontend/src/components/EditUser/index.tsx [hunk 1]
```diff
@@ -81,10 +101,46 @@ class EditUser extends React.Component<Props, State> {
...
         invalidPassword: this.state.password.length > 0 && this.state.password.length < 8,
         mismatchPasswords: this.state.password !== this.state.confirmPassword
+      },
+      () => {
+        if (this.state.missingName
+          || this.state.invalidEmail
+          || this.state.missingPassword
+          || this.state.invalidPassword
+          || this.state.mismatchPasswords) {
+            return
```
Comment (KentShikama) [inline src/frontend/src/components/EditUser/index.tsx]: We probably should just invert this if statement 

`if !(...) { this.dispatchEditUser(); }`

Comment (grbarker) [reply src/frontend/src/components/EditUser/index.tsx]: I changed it to remove the else statement.

File: src/frontend/src/components/EditUser/index.tsx [hunk 2]
```diff
@@ -81,10 +101,44 @@ class EditUser extends React.Component<Props, State> {
         missingPassword: this.state.password.length === 0,
         invalidPassword: this.state.password.length > 0 && this.state.password.length < 8,
         mismatchPasswords: this.state.password !== this.state.confirmPassword
+      },
+      () => {
+        if (!this.state.missingName
```
[the remark this answers is not in the document]
Comment (grbarker) [reply src/frontend/src/components/EditUser/index.tsx]: That is the same suggestion Kent had, but it is incorrect syntax. The app doesn't compile for me and explicitly points to that ! as the cause of the problem.

[the remark this answers is not in the document]
Comment (grbarker) [reply src/frontend/src/components/EditUser/index.tsx]: To get the same removal of the else statment I put it in front of all the error checks and switched the operators over to "&&".

[the remark this answers is not in the document]
Comment (wittejm) [reply src/frontend/src/components/EditUser/index.tsx]: ah, gotcha. I think some extra parens would work?
```
if (
    !(this.state.missingName
          || this.state.invalidEmail
          || this.state.missingPassword
          || this.state.invalidPassword
          || this.state.mismatchPasswords
    )
  ) {
            this.dispatchEditUser();
          }
```

[the remark this answers is not in the document]
Comment (grbarker) [reply src/frontend/src/components/EditUser/index.tsx]: I looked at the backend and it seems to require a password be a part of the request data.
`if not any([key in data.keys() for key in ["email", "name", "group_name", "password", "admin"]]):
        error(
            400,
            "Json data must define one or more of: \
    email, name, group_name, password, admin",
        )`

[the remark this answers is not in the document]
Comment (wittejm) [reply src/frontend/src/components/EditUser/index.tsx]: (I answered this offline but for anyone else reading this:) It just needs one of those fields in order to pass.

**Comment by wittejm** [comment]
The edit attempt still fails if the password is left empty. It looks like the `password` field is still being passed as an empty string in the request, so the backend returns the 422 error `New password is less than 8 characters long`. The way to get around that would be to exclude the `password` field entirely from the request if the field is empty.

**Comment by wittejm** [comment]
Changes look good apart from new set of redundant git history

**Pull request merged**